[WIP] データベース構造の設計

miya98935271 opened the pull request:
#What
マークダウン記法を用いてReadme.mdにデータベース構造を記述する。

#Why
正しく正規化された、メンテナンス性の高いデータベースを作成するため。

Commit:
create pull request

Commit:
Database Table design

Database markdown to Readme.md

Commit:
Revert "Database Table design"

This reverts commit e75baf514bc1373721333f1d9ac5e946a5fc33f2.

Commit:
write Database design

write Database design to Readme.md

i-date commented on README.md:
```diff
@@ -1,24 +1,49 @@
+|Column|Type|Options|
+|------|----|-------|
+|user_id|integer|null: false, foreign_key: true|
+|group_id|integer|null: false, foreign_key: true|
```
6-7行目の外部キーは、Railsではinteger型ではなく、references型を使用しましょう。
そうでない場合、`foreign_key: true`では外部キー制約にならないので注意です。
また、そうすることで `user_id` を `user` と記述するとテーブルが生成される時に _id が付与されます。
例：`|user|references|null: false, foreign_key: true|`
あと他の外部キーも修正しましょう。
参考：https://qiita.com/ryouzi/items/2682e7e8a86fd2b1ae47

miya98935271 replied:
参考ページの提供ありがとうございます。
部キー使用の部分を修正します。
~~~
|Column|Type|Options|
|------|----|-------|
|user|references|null: false, foreign_key: true|
|group|references|null: false, foreign_key: true|
~~~

i-date commented on README.md:
```diff
@@ -1,24 +1,49 @@
+## usersテーブル
+|Column|Type|Options|
+|------|----|-------|
+|username|string|null: false|
```
usernameではなく、nameにしましょう。
Columnはテーブル名のusers（というエンティティ）の属性を表すので、わざわざuserを付ける必要はありません。usersテーブルのnameでユーザー名と十分わかるでしょう。

また、usersテーブルのnameカラムはユーザー検索に使用するので、Optionsに`index: true`を追加しましょう。

miya98935271 replied:
ありがとうございます。
当該箇所の修正します。
~~~
|name|string|null: false, index: true|
~~~

i-date commented on README.md:
```diff
@@ -1,24 +1,49 @@
+|------|----|-------|
+|username|string|null: false|
+|useremail|string|null: false|
+|userpassword|string|null: false|
```
18-19行目はdeviseにより自動生成されるので記述しなくても構いません。

miya98935271 replied:
了解です。
修正します。
~~~
|------|----|-------|
|name|string|null: false, index: true|
~~~

i-date commented on README.md:
```diff
@@ -1,24 +1,49 @@
+|Column|Type|Options|
+|------|----|-------|
+|title|string|null: false|
+|user_id|integer|null: false, foreign_key: true|
```
31行目は不要ですので削除しましょう。

miya98935271 replied:
了解です、修正します。
グループはそれを作成したユーザーに紐つくものだと仕様を誤解していたためユーザー情報を組み込んでいました。
~~~
|Column|Type|Options|
|------|----|-------|
|title|string|null: false|
~~~

i-date commented on README.md:
```diff
@@ -1,24 +1,49 @@
+|user_id|integer|null: false, foreign_key: true|
+
+### Association_users
+- has_many :users, through: members
```
34行目の`members`の前に`:`を挿れましょう。
`has_many :users, through: :members`

miya98935271 replied:
了解しました。
修正します。
~~~
### Association_users
- has_many :users, through: :members
~~~

i-date commented on README.md:
```diff
@@ -1,24 +1,49 @@
+|Column|Type|Options|
+|------|----|-------|
+|comment|text|null: false|
+|image|mediumblob|
```
commemtとimageはどちらも単体で保存できるようにするため、`null: false`は削除しましょう。
また、imageはstring型が一般的です。

miya98935271 replied:
imageカラムについて、修正します。
~~~
|image|string|
~~~

42行目について理解ができてないのですが、
まず、chat-spaceで空のメッセージを入力した際に"メッセージを入力してください"とのウィンドウメッセージが表示されました。
このためメッセージが入るcommentカラムに"null: false"が定められていると考えたのですが違うのでしょうか？
もしかして別の部分で空メッセージを弾く処理をしているのですか？

i-date commented on README.md:
```diff
@@ -1,24 +1,49 @@
+|comment|text|null: false|
+|image|mediumblob|
+|user_id|integer|null: false, foreign_key: true|
+|group_id|integer|null: false, foreign_key: true|
```
44-45行目もinteger型ではなく、references型を使用しましょう。

miya98935271 replied:
了解です、修正します。
~~~
|user|references|null: false, foreign_key: true|
|group|references|null: false, foreign_key: true|
~~~

i-date reviewed (commented):
ご提出ありがとうございます！コメントいたしました^^
また、修正したら再度レビューを依頼しましょう。

Commit:
resolve problems

- correct column name
- correct integer types to references types using foreign key
- correct types of image column to string
- correct lack of colon
- delete options of comment column
- delete unnecessary column

TOMARUMARU reviewed (approved):
LGTM
